### ISSUE TITLE
Ethers 4.0.43 breaks type generation

### ISSUE THREAD

**Issue by HenryNguyen5** [ethereum-ts/TypeChain#206]
Upgrading our codebase from ethers 4.0.41 to 4.0.43 breaks the generated types during compilation.

```ts

src/generated/BasicConsumerFactory.ts:13:11 - error TS2345: Argument of type '({ constant: boolean; inputs: { name: string; type: string; }[]; name: string; outputs: never[]; payable: boolean; stateMutability: string; type: string; anonymous?: undefined; } | { constant: boolean; inputs: never[]; ... 5 more ...; anonymous?: undefined; } | { ...; } | { ...; })[]' is not assignable to parameter of type 'string | (string | FunctionFragment | EventFragment)[] | Interface'.
  Type '({ constant: boolean; inputs: { name: string; type: string; }[]; name: string; outputs: never[]; payable: boolean; stateMutability: string; type: string; anonymous?: undefined; } | { constant: boolean; inputs: never[]; ... 5 more ...; anonymous?: undefined; } | { ...; } | { ...; })[]' is not assignable to type '(string | FunctionFragment | EventFragment)[]'.
    Type '{ constant: boolean; inputs: { name: string; type: string; }[]; name: string; outputs: never[]; payable: boolean; stateMutability: string; type: string; anonymous?: undefined; } | { constant: boolean; inputs: never[]; ... 5 more ...; anonymous?: undefined; } | { ...; } | { ...; }' is not assignable to type 'string | FunctionFragment | EventFragment'.
      Type '{ inputs: { name: string; type: string; }[]; payable: boolean; stateMutability: string; type: string; constant?: undefined; name?: undefined; outputs?: undefined; anonymous?: undefined; }' is not assignable to type 'string | FunctionFragment | EventFragment'.
        Type '{ inputs: { name: string; type: string; }[]; payable: boolean; stateMutability: string; type: string; constant?: undefined; name?: undefined; outputs?: undefined; anonymous?: undefined; }' is not assignable to type 'FunctionFragment'.
          Types of property 'name' are incompatible.
            Type 'undefined' is not assignable to type 'string'.

13     super(_abi, _bytecode, signer);
             ~~~~

src/generated/BasicConsumerFactory.ts:40:34 - error TS2345: Argument of type '({ constant: boolean; inputs: { name: string; type: string; }[]; name: string; outputs: never[]; payable: boolean; stateMutability: string; type: string; anonymous?: undefined; } | { constant: boolean; inputs: never[]; ... 5 more ...; anonymous?: undefined; } | { ...; } | { ...; })[]' is not assignable to parameter of type 'string | (string | FunctionFragment | EventFragment)[] | Interface'.
  Type '({ constant: boolean; inputs: { name: string; type: string; }[]; name: string; outputs: never[]; payable: boolean; stateMutability: string; type: string; anonymous?: undefined; } | { constant: boolean; inputs: never[]; ... 5 more ...; anonymous?: undefined; } | { ...; } | { ...; })[]' is not assignable to type '(string | FunctionFragment | EventFragment)[]'.

40     return new Contract(address, _abi, signerOrProvider) as BasicConsumer;

```

**Comment by kermankohli** [comment]
Facing this issue as well. Have you solved it? If so what's the correct combo of Ethers version and Typechain version? Been stuck on this for a day now!

**Comment by krzkaczor** [comment]
Sorry for that @kermankohli . For now, workaround would be not to upgrade to `ethers@4.0.43`. I will try to release a fix this week.

FYI: @ricmoo I think there was a breaking change in ethers typings. Would be cool if you could semver them as well 🤔

**Comment by ricmoo** [comment]
That change should not have been breaking, since anyone passing in an object of that type would have crashed 2 lines later, but I will put ParamType back into the typing, and rely on runtime errors, which is no worse than previous behaviour.

Sorry for the inconvenience.

**Comment by HenryNguyen5** [comment]
Thanks @ricmoo @krzkaczor!

**Comment by HenryNguyen5** [comment]
@krzkaczor should we close this since it should be fixed upstream?

**Comment by HenryNguyen5** [comment]
It's published under 4.0.44 now, maybe you could add it as a restriction under peerdeps

**Comment by krzkaczor** [comment]
Thanks for notifying me @HenryNguyen5 I'll just close it.
